### PR TITLE
Fixes related to pagination constraints

### PR DESCRIPTION
**Pull request by SebaKas21**
Hi @dawiddworak88 

I think I need to remove unnecessary validators in controllers, what do you think?

### CODE REVIEW

**Comment by dawiddworak88** [comment]
Yes :)

W dniu pt., 2.12.2022 o 13:37 Sebastian Kasperek ***@***.***>
napisał(a):

File: be/src/Infrastructure/GenericRepository/Foundation.GenericRepository/Definitions/Constants.cs [hunk 1]
```diff
@@ -5,5 +5,7 @@ public static class Constants
         public static readonly int DefaultPageIndex = 1;
         public static readonly int DefaultItemsPerPage = 25;
         public static readonly int MaxItemsPerPage = 100;
+        public static readonly int MaxItemsPerPageLimit = 1000;
```
Comment (dawiddworak88) [inline be/src/Infrastructure/GenericRepository/Foundation.GenericRepository/Definitions/Constants.cs]: **suggestion:** let's change it to 2000

File: be/src/Infrastructure/GenericRepository/Foundation.GenericRepository/Definitions/Constants.cs [hunk 2]
```diff
@@ -5,5 +5,7 @@ public static class Constants
...
+        public static readonly int EmptyTotal = 0;
```
Comment (dawiddworak88) [inline be/src/Infrastructure/GenericRepository/Foundation.GenericRepository/Definitions/Constants.cs]: **suggestion:** let's remove it completely

File: be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs [hunk 3]
```diff
@@ -51,7 +52,18 @@ public async Task<PagedResults<IEnumerable<CategoryServiceModel>>> GetAsync(GetC
 
             categories = categories.ApplySort(model.OrderBy);
 
-            var pagedResults = categories.PagedIndex(new Pagination(categories.Count(), model.ItemsPerPage), model.PageIndex);
+            var pagedResults = categories.PagedIndex(new Pagination(Constants.EmptyTotal, Constants.MaxItemsPerPageLimit), Constants.DefaultPageIndex);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs]: **suggestion:** please replace with the following everywhere:

`PagedResults<IEnumerable<Category>> pagedResults;`

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

File: be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs (hunk 3 above)
Comment (SebaKas21) [reply be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs]: Hi @dawiddworak88 as variable?

File: be/src/Project/Services/Client/Client.Api/v1/Controllers/ClientsApplicationsController.cs [hunk 4]
```diff
@@ -290,41 +290,35 @@ public async Task<IActionResult> Get(string ids, string searchTerm, int pageInde
...
+                        })
+                    };
 
-                        return this.StatusCode((int)HttpStatusCode.OK, response);
-                    }
+                    return this.StatusCode((int)HttpStatusCode.OK, response);
                 }
 
-                throw new CustomException(string.Join(ErrorConstants.ErrorMessagesSeparator, validationResult.Errors.Select(x => x.ErrorMessage)), (int)HttpStatusCode.UnprocessableEntity);
+                throw new CustomException("", (int)HttpStatusCode.UnprocessableEntity);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Client/Client.Api/v1/Controllers/ClientsApplicationsController.cs]: **suggestion:** let's always simply return  this.StatusCode((int)HttpStatusCode.UnprocessableEntity); if there is no error message.

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]
We're almost there ⭐

**Review by dawiddworak88 — APPROVED** [review]
Thank you 🥇

**Pull request merged**